ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -1450,6 +1465,17 @@ ORK_CLASS_AVAILABLE
 ORK_CLASS_AVAILABLE
 @interface ORKEmailAnswerFormat : ORKAnswerFormat
 
+/**
+ Identifies whether this email answer format is being used as a username.
+ 
+ For integration with iOS 12's password management functionality. Use this answer format if your
+ username is also an email address, if it is not guaranteed to be an email address, use
+ `ORKTextAnswerFormat` and set the `textContentType` to `UITextContentTypeUsername`.
+ 
+ By default, the value of this property is NO.
+ */
+@property (nonatomic,getter=isUsernameField) BOOL usernameField;
+
```

Comment:
@edelabar I am struggling a bit to understand the need for this property - shouldn't a developer use an ORKTextAnswerFormat if they anticipate the user to provide a username and only use ORKEmailAnswerFormat if the user is expected to provide their email address? Why do you want to support this exception? Isn't it better to keep them distinct? 

Would love to hear your thoughts on this..

Reply:
The password management feature stores username/password pairs in the keychain. Some apps use a text username (e.g. edelabar) others use an email address. 

During registration, if your app needs an email address as a username, you can use ORKEmailAnswerFormat to validate that the username is an email address, and use the usernameField property to set the textContentType to UITextContentTypeUsername so it's stored in the keychain appropriately. 

If your system's username does not need to be an email address, then an ORKTextAnswerFormat with textContentType set to UITextContentTypeUsername is fine.

Reply:
(For a login screen an ORKTextAnswerFormat with textContentType set to UITextContentTypeUsername is probably correct regardless of whether or not an email address is required as a username, this is really just a cse for registration.)

Reply:
Got it, thanks for the clarification.